### PR TITLE
Plugin/content manager dev

### PR DESCRIPTION
**Pull request by Aurelsicoko**

### CODE REVIEW

File: packages/strapi-helper-plugin/lib/src/components/PopUpWarning/tests/index.test.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// import PopUpWarning from '../index';
+
+import expect from 'expect';
+// import { shallow } from 'enzyme';
```
Comment (pierreburgy) [inline packages/strapi-helper-plugin/lib/src/components/PopUpWarning/tests/index.test.js]: Remove useless comments

Comment (pierreburgy) [reply packages/strapi-helper-plugin/lib/src/components/PopUpWarning/tests/index.test.js]: Actually, remove every `tests` folders because they are not used.

File: packages/strapi-helper-plugin/lib/src/utils/cleanData.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { isArray } from 'lodash';
```
Comment (pierreburgy) [inline packages/strapi-helper-plugin/lib/src/utils/cleanData.js]:
```js
import { isArray, isObject } from 'lodash';

const cleanData = (value, key, secondKey) => {
  if (isArray(value)) {
    return value.map(obj => (obj[key] ? obj[key] : key));
  } else if (isObject(value)) {
    return value[key] || value[secondKey];
  } else {
    return value;  
  }
}

export default cleanData;
```

File: packages/strapi-helper-plugin/lib/src/utils/request.js [hunk 3]
```diff
@@ -8,7 +8,11 @@ import 'whatwg-fetch';
  * @return {object}          The parsed JSON from the request
  */
 function parseJSON(response) {
-  return response.json();
+  if (response.json) {
```
Comment (pierreburgy) [inline packages/strapi-helper-plugin/lib/src/utils/request.js]:
```js
return response.json ? response.json() : response;
```

File: packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js [hunk 4]
```diff
@@ -4,131 +4,65 @@
...
   }
-  /**
-   * Triggered on dots click
-   *
-   * Simply prevent link default behavior
-   *
-   * @param e {Object} Click event
-   */
-  onDotsClicked(e) {
+
```
Comment (pierreburgy) [inline packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js]: Why did you remove the comments?

File: packages/strapi-plugin-content-manager/admin/src/components/SelectMany/index.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            value: item,
+            label: templateObject({ mainField: this.props.relation.displayedAttribute }, item).mainField,
+          })) :
+          [{
+            value: response,
+            label: response[this.props.relation.displayedAttribute],
+          }];
+
+        return { options };
+      });
```
Comment (pierreburgy) [inline packages/strapi-plugin-content-manager/admin/src/components/SelectMany/index.js]: It would be great to manage errors here.

File: packages/strapi-plugin-content-manager/admin/src/components/SelectOne/index.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+          map(response, item => ({
+            value: item,
+            label: templateObject({ mainField: this.props.relation.displayedAttribute }, item).mainField,
+          })) :
+          [{
+            value: response,
+            label: templateObject({ mainField: this.props.relation.displayedAttribute }, response).mainField,
+          }];
+
+        return {options};
```
Comment (pierreburgy) [inline packages/strapi-plugin-content-manager/admin/src/components/SelectOne/index.js]: It would be great to manage errors here.

File: packages/strapi-plugin-content-manager/admin/src/containers/App/sagas.js [hunk 7]
```diff
@@ -1,14 +1,34 @@
...
+      .then(response => generateSchema(response))
+      .then(displayedModels => {
+        return [{
+          name: 'Content Types',
+          links: _.map(displayedModels, (model, key) => ({
+            label: model.labelPlural || model.label || key,
+            destination: key,
+          })),
+        }];
+      });
```
Comment (pierreburgy) [inline packages/strapi-plugin-content-manager/admin/src/containers/App/sagas.js]: Catch should be used directly in the promise (`.catch`), not using a try/catch.

File: packages/strapi-plugin-content-manager/admin/src/containers/Edit/sagas.js [hunk 8]
```diff
@@ -41,21 +41,27 @@ export function* editRecord() {
...
     yield call(request, requestUrl, {
       method: isCreating ? 'POST' : 'PUT',
-      body: recordJSON,
+      body: recordCleaned,
     });
 
     yield put(recordEdited());
     window.Strapi.notification.success(
-      `The entry has been successfully ${isCreating ? 'created' : 'updated'}.`
+      `Saved`
```
Comment (pierreburgy) [inline packages/strapi-plugin-content-manager/admin/src/containers/Edit/sagas.js]: This message should be translated.

File: packages/strapi-plugin-content-manager/admin/src/containers/Edit/sagas.js [hunk 9]
```diff
@@ -65,49 +71,54 @@ export function* editRecord() {
...
+      const requestUrl = `${window.Strapi.apiUrl}/content-manager/explorer/${modelName}/${id}`;
+
+      // Call our request helper (see 'utils/request')
+      yield call(request, requestUrl, {
+        method: 'DELETE',
+      });
+
+      yield put(recordDeleted(id));
+      window.Strapi.notification.success(
+        'Deleted'
```
Comment (pierreburgy) [inline packages/strapi-plugin-content-manager/admin/src/containers/Edit/sagas.js]: This message should be translated.

File: packages/strapi-plugin-content-manager/admin/src/containers/Edit/sagas.js [hunk 10]
```diff
@@ -65,49 +71,54 @@ export function* editRecord() {
...
+      );
+
+      // Redirect to the list page.
+      router.push(`/plugins/content-manager/${modelName}`);
+    } catch (err) {
+      yield put(recordDeleteError());
+      window.Strapi.notification.error(
+        'An error occurred during record deletion.'
```
Comment (pierreburgy) [inline packages/strapi-plugin-content-manager/admin/src/containers/Edit/sagas.js]: This message should be translated.

File: packages/strapi-plugin-content-manager/admin/src/containers/App/sagas.js [hunk 11]
```diff
@@ -1,14 +1,34 @@
...
+  } catch (err) {
+    window.Strapi.notification.error(
+      'An error occurred during models config fetch.'
```
Comment (pierreburgy) [inline packages/strapi-plugin-content-manager/admin/src/containers/App/sagas.js]: This message should be translated.

File: packages/strapi-admin/admin/src/containers/NotFoundPage/index.js [hunk 12]
```diff
@@ -28,8 +29,24 @@ export default class NotFound extends React.Component { // eslint-disable-line r
...
+        <Button
+          label="Back to home page"
+          buttonBackground="back"
+          onClick={(e) => {
+            e.stopPropagation();
+
+            this.props.history.goBack();
+          }}
+        >
+          Back to home page.
```
Comment (pierreburgy) [inline packages/strapi-admin/admin/src/containers/NotFoundPage/index.js]: `Button` component does not require content, because it is managed by the `label` prop.

File: packages/strapi-admin/admin/src/containers/NotFoundPage/index.js [hunk 13]
```diff
@@ -28,8 +29,24 @@ export default class NotFound extends React.Component { // eslint-disable-line r
         <h2 className={styles.notFoundDescription}>
           <FormattedMessage {...messages.description} />
         </h2>
-        <Link to="/admin">Back to home page.</Link>
+        <Button
+          label="Back to home page"
```
Comment (pierreburgy) [inline packages/strapi-admin/admin/src/containers/NotFoundPage/index.js]: The label should be translated (pass the translation id as label and add `handlei18n` prop).

File: packages/strapi-admin/admin/src/components/NotificationsContainer/styles.scss [hunk 14]
```diff
@@ -3,8 +3,8 @@
 
 .notificationsContainer { /* stylelint-disable */
   position: absolute;
-  top: 7rem;
-  right: 1rem;
+  top: 72px;
```
[the remark this answers is not in the document]
Comment (Aurelsicoko) [reply packages/strapi-admin/admin/src/components/NotificationsContainer/styles.scss]: It depends... It's recommended for font-size, margin and spacing. For the other properties such as width, height, top, left, etc..., I've read both bits of advice on these ones.

File: packages/strapi-helper-plugin/lib/src/components/Button/styles.scss [hunk 15]
```diff
@@ -0,0 +1,95 @@
+.button {
+  -webkit-font-smoothing: antialiased;
```
[the remark this answers is not in the document]
Comment (Aurelsicoko) [reply packages/strapi-helper-plugin/lib/src/components/Button/styles.scss]: No. It's already the case, but for some icons and fonts, we have to change the value to improve the rendering.

File: packages/strapi-helper-plugin/lib/src/components/Input/styles.scss [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+      width: 32px;
+      height: 32px;
+      background: #FAFAFB;
+      border-radius: 3px 0px 0px 3px;
+      font-family: 'FontAwesome';
+      font-size: 1.4rem;
+      line-height: 32px;
+      text-align: center;
+      color: #B3B5B9;
+      -webkit-font-smoothing: none;
```
[the remark this answers is not in the document]
Comment (Aurelsicoko) [reply packages/strapi-helper-plugin/lib/src/components/Input/styles.scss]: Yes.

File: packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js (hunk 4 above)
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js]: Because I thought that the function naming was good enough to understand what the function does

File: packages/strapi-helper-plugin/lib/src/utils/cleanData.js (hunk 2 above)
Comment (Aurelsicoko) [reply packages/strapi-helper-plugin/lib/src/utils/cleanData.js]: Thanks for this one. A mistake though... 

```js
...
return value.map(obj => (obj[key] ? obj[key] : obj)); // We have to return the obj not the key :)
...
```

File: packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js (hunk 4 above)
Comment (Aurelsicoko) [reply packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js]: @soupette ?

Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js]: Well since the naming is explicit and the code within it is easy to understand I thought that the comments were useless...

File: packages/strapi-helper-plugin/lib/src/components/Button/styles.scss (hunk 15 above)
[the remark this answers is not in the document]
Comment (pierreburgy) [reply packages/strapi-helper-plugin/lib/src/components/Button/styles.scss]: Alright, thanks.

File: packages/strapi-helper-plugin/lib/src/components/Input/styles.scss (hunk 16 above)
[the remark this answers is not in the document]
Comment (pierreburgy) [reply packages/strapi-helper-plugin/lib/src/components/Input/styles.scss]: Ok, so keep this line like it is.

File: packages/strapi-helper-plugin/lib/src/utils/cleanData.js (hunk 2 above)
Comment (pierreburgy) [reply packages/strapi-helper-plugin/lib/src/utils/cleanData.js]: Good catch, sorry for the mistake.

File: packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js (hunk 4 above)
Comment (pierreburgy) [reply packages/strapi-plugin-content-manager/admin/src/components/Pagination/index.js]: Comments can be useful to hep new comers understand what is the logic and the choices behind the functions.

File: packages/strapi-helper-plugin/lib/src/components/Input/styles.scss [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+  text-transform: capitalize;
+  -moz-appearance: none;
+  -webkit-appearance: none;
+}
+
+.input {
+  min-width: 200px;
+  margin-bottom: 1.5rem;
+  font-size: 1.3rem;
+  font-family: 'Lato';
```
[the remark this answers is not in the document]
Comment (Aurelsicoko) [reply packages/strapi-helper-plugin/lib/src/components/Input/styles.scss]: Our override of Bootstrap isn't perfect. This is a part of what we are to improve during the next months. Right now, the goal is to ensure that the display is the right one. Then, we could improve the number of properties used to reach this result.
